pkg/utils/utils.go

```diff
@@ -287,12 +287,15 @@ func configSriovDevice(iface *sriovnetworkv1.Interface, ifaceStatus *sriovnetwor
 					break
 				}
 			}
-			if strings.EqualFold(iface.LinkType, "IB") {
-				if err = setVfGuid(addr, pfLink); err != nil {
+			if iface.NumVfs != ifaceStatus.NumVfs {
+				// only set VF GUID and MAC for when VFs are created
+				if strings.EqualFold(iface.LinkType, "IB") {
+					if err = setVfGuid(addr, pfLink); err != nil {
+						return err
+					}
+				} else if err = setVfAdminMac(addr, pfLink); err != nil {
 					return err
 				}
-			} else if err = setVfAdminMac(addr, pfLink); err != nil {
-				return err
 			}
 			if err = unbindDriverIfNeeded(addr, isRdma); err != nil {
```

Comment:
Put `unbindDriverIfNeeded` in the same conditional check (line 290)?

Reply:
why do you want to pull `unbindDriverIfNeeded` into the new condition? I think this might lead to new unwanted cases, i.e. why you should only unbind when VF num changed? Since this function is not dependent on any module to be loaded I would keep it as is.

Reply:
It is because `unbindDriverIfNeeded` is a workaround for infiniband (IPoIB) case (`isRdma` == true) to set GUID,
but when `isRdma` == true, user may just want to use dpdk with Mellanox device (in bifurcated mode), in which case, the unbind is not needed even `isRdma` equals to true.

Reply:
In other words, `unbindDriverIfNeeded` is only needed when initializing(vf num changed) infiniband device.